public/lib/pdf_from_dcs.py

```diff
@@ -177,78 +257,131 @@ def create_pdf(self, obs_obj: OBS) -> str:
 
             except subprocess.CalledProcessError as e:
+                self.output_msg += f"{datetime.datetime.now()} => ConTeXt process failed!\n"
+                write_file(self.output_msg_filepath, self.output_msg)
 
                 # find the tex error lines
                 std_out = e.stdout.decode('utf-8')
                 std_out = re.sub(r'\n\n+', '\n', std_out, flags=re.MULTILINE)
                 err_lines = re.findall(r'(^tex error.+)\n?', std_out, flags=re.MULTILINE)
 
                 write_file(out_log, std_out)
-                write_file(err_log, '\n'.join(err_lines))
+                write_file(err_log_path, '\n'.join(err_lines))
 
-                raise ChildProcessError('Errors were generated by ConTeXt. See {}.'.format(err_log))
+                err_msg = f"Errors were generated by ConTeXt. See {err_log_path}."
+                self.output_msg += f"{datetime.datetime.now()} ERROR: {err_msg}\n"
+                write_file(self.output_msg_filepath, self.output_msg)
+                raise ChildProcessError(err_msg)
 
-            self.output += str(datetime.datetime.now()) + ' => Copying the PDF file to output directory.\n'
+            # PDF file is in out_dir
+            pdf_current_filepath = os.path.join(out_dir, f'{obs_lang_code}.pdf')
             version = obs_obj.version.replace('.', '_')
             if version[0:1] != 'v':
-                version = 'v' + version
+                version = f'v{version}'
+            pdf_desired_name = f'obs-{obs_lang_code}-{version}.pdf'
+
+            # Copy the new PDF file to the /app/obs-pdf/output/{obs_lang_code}/ folder
+            # self.output_msg += f"{datetime.datetime.now()} => Copying the '{obs_lang_code}' PDF file to output directory…\n"
+            # write_file(self.output_msg_filepath, self.output_msg)
+            # output_dir = os.path.join(get_output_dir(), obs_lang_code)
+            # if not isdir(output_dir):
+            #     make_dir(output_dir, linux_mode=0o777, error_if_not_writable=True)
+            # pdf_destination_filepath = os.path.join(output_dir, pdf_desired_name)
+            # self.output_msg += f"  Copying {pdf_current_filepath} to {pdf_destination_filepath}…\n"
+            # write_file(self.output_msg_filepath, self.output_msg)
+            # shutil.copyfile(pdf_current_filepath, pdf_destination_filepath)
+
+            # Upload the PDF to our AWS S3 bucket
+            self.output_msg += f"{datetime.datetime.now()} => Uploading the '{obs_lang_code}' PDF file to S3 bucket…\n"
+            write_file(self.output_msg_filepath, self.output_msg)
+            cdn_s3_handler = S3Handler(bucket_name=self.cdn_bucket_name,
+                                       aws_access_key_id=self.aws_access_key_id,
+                                       aws_secret_access_key=self.aws_secret_access_key,
+                                       aws_region_name=self.aws_region_name)
+            s3_commit_key = f'obs/auto_PDFs/{pdf_desired_name}'
```

Comment:
Per slack thread, don't put them here.  Use Option 2 listed in https://github.com/unfoldingWord-dev/door43.org/issues/389